modules/geolocation/geolocation.py

```diff
@@ -96,6 +95,77 @@ def gather_point_pairs(self):
             pixelGeoPairs = np.concatenate((pixelGeoPairs, [pair]))
 
         return pixelGeoPairs
+    
+    def __are_three_points_collinear(self, p1, p2, p3):
+        """
+        PRIVATE
+        Evaluates whether the three given points are collinear
+
+        Parameters
+        ----------
+        p1 : tuple/np.ndarray/list
+            Point 1, in form (x, y)
+        p2 : tuple/np.ndarray/list
+            Point 2, in form (x, y)
+        p3 : tuple/np.ndarray/list
+            Point 3, in form (x, y)
+
+        Returns
+        -------
+        bool
+            True if the three points are collinear, otherwise false
+        """
+        x = 0
+        y = 1
+        return 0.5*(p1[x]*(p2[y]-p3[y]) +
+                    p2[x]*(p3[y]-p1[y]) +
+                    p3[x]*(p1[y]-p2[y])) == 0
+
+
+    def get_non_collinear_points(self, coordinatesArray):
+        """
+        Returns a list of four coordinates from an input array that are not collinear to one another.
+
+        Parameters
+        ----------
+        coordinatesArray : np.ndarray
+            Array with dimensions ( , 2), containing a list of coordinates
+
+        Returns
+        -------
+        np.ndarray
+            Array with dimensions (4, 2), containing a list of coordinates that are non-collinear,
+            or an empty list if none were found in the input array
+        """
+        # Empty array in case no set of four non-collinear points are found
+        collinearPoints = np.zeros(shape=(4, 2))
+
+        # If there aren't four points, return the empty array
+        if len(coordinatesArray) < 4:
+            return collinearPoints
+
+        # Look at all sequential pairs of four points
+        for i in range(0, len(coordinatesArray)):
```

Comment:
I can make this algorithm fail, but the circumstances in which it does is even less likely than my original one. It is slightly slower (not really an issue given how few points it will be called with, but just a consideration).

Add a comment explaining why sequential (rather than all combinations).

(Not part of PR) Note: I believe an actual correct algorithm must at minimum check all combinations, which means the problem is NP (solvable in exponential-time, verifiable in polynomial-time).

Reply:
That is to say, add the comment, no need to change anything.